From ef9c36d4b525ced08fe595a3e6eeaa0d1e21ef88 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:26:34 -0400
Subject: [tools] perf/sched: Optimize build time

Message-id: <1408462094-14194-226-git-send-email-jolsa@redhat.com>
Patchwork-id: 88253
O-Subject: [PATCH RHEL7.1 BZ1131394 225/325] perf sched: Optimize build time
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 156a2b022907687f28c72d1ba601015f295cd99e
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Tue Oct 22 10:34:16 2013 +0300

description
===========
builtin-sched.c took a log time to build with -O6 optimization. This
turned out to be caused by:

 .curr_pid = { [0 ... MAX_CPUS - 1] = -1 },

Fix by initializing curr_pid programmatically.

This addresses the problem cured in f36f83f947ed using a smaller hammer.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-sched.c b/tools/perf/builtin-sched.c
index 5a33856..ddb5dc1 100644
--- a/tools/perf/builtin-sched.c
+++ b/tools/perf/builtin-sched.c
@@ -1670,7 +1670,6 @@ int cmd_sched(int argc, const char **argv, const char *prefix __maybe_unused)
   .sort_list       = LIST_HEAD_INIT(sched.sort_list),
   .start_work_mutex     = PTHREAD_MUTEX_INITIALIZER,
   .work_done_wait_mutex = PTHREAD_MUTEX_INITIALIZER,
-  .curr_pid       = { [0 ... MAX_CPUS - 1] = -1 },
   .sort_order       = default_sort_order,
   .replay_repeat       = 10,
   .profile_cpu       = -1,
@@ -1732,6 +1731,10 @@ int cmd_sched(int argc, const char **argv, const char *prefix __maybe_unused)
   .switch_event     = replay_switch_event,
   .fork_event     = replay_fork_event,
  };
+ unsigned int i;
+
+ for (i = 0; i < ARRAY_SIZE(sched.curr_pid); i++)
+  sched.curr_pid[i] = -1;
 
  argc = parse_options(argc, argv, sched_options, sched_usage,
         PARSE_OPT_STOP_AT_NON_OPTION);
-- 
1.7.1